python-package/lightgbm/sklearn.py

```diff
@@ -847,6 +847,20 @@ def objective_(self):
             raise LGBMNotFittedError('No objective found. Need to call fit beforehand.')
         return self._objective
 
+    @property
+    def n_estimators_(self):
+        """:obj:`int`: The concrete number of boosted trees this model contains after fitting."""
```

Comment:
I think users who are not very familiar with LightGBM specifically and / or GBDTs generally might not understand what "concrete number of boosted trees" means, or why it would be different from parameter `num_iterations` and its aliases.

I also think that phrase could be misinterpreted in the case of multi-class classification, where LightGBM fits one tree per class and the total number of *trees* is actually `n_classes * n_iterations`.

Would you consider something like this, here and for `n_iter_`?

> True number of boosting iterations performed. This might be less than parameter `num_iterations` if early stopping was enabled or if boosting stopped early due to limits on complexity like `min_gain_to_split`.

The `min_gain_to_split` comment is a reference to https://github.com/microsoft/LightGBM/blob/9e1d7fa1bb395a9c4e5f7af9b13b9eb315e360fa/src/boosting/gbdt.cpp#L441

---------

Also want to note here that the `sklearn.ensemble.GradientBoostingClassifier.n_estimators_` doc also references early stopping explicitly.

https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingClassifier.html

Reply:
Thank you! I agree with the proposed wording.
I believe that more useful description is better than keeping consistent with scikit-learn's wording for `n_iter_`.